api/v1beta1/hcpvaultsecretsapp_types.go

```diff
@@ -32,6 +32,37 @@ type HCPVaultSecretsAppSpec struct {
 	// Destination provides configuration necessary for syncing the HCP Vault
 	// Application secrets to Kubernetes.
 	Destination Destination `json:"destination"`
+	// SyncConfig configures sync behavior from HVS to VSO
+	SyncConfig *HVSSyncConfig `json:"syncConfig,omitempty"`
+}
+
+// HVSSyncConfig configures sync behavior from HVS to VSO
+type HVSSyncConfig struct {
+	// Dynamic configures sync behavior for dynamic secrets.
+	Dynamic *HVSDynamicSyncConfig `json:"dynamic,omitempty"`
+}
```

Comment:
I debated whether to reuse the existing SyncConfig{} type from VaultStaticSecret, but since HVS is a bit different than Vault I went with a new type here but still named "SyncConfig" in the HVSApp spec.

Reply:
I think new `HVSSyncConfig` for HVS makes sense.